examples/timeSeries-hand-gestures/sketch.js

```diff
@@ -0,0 +1,217 @@
+      if (train_word.length > 0){
+        // add data to the model
+        let target = {label:train_word}
+        model.addData(sequence, target);
+        trainingWordsUpdate()
+
+      // if there is no word in the box then classify instead
+      } else {
+        // classify the data
+        model.classify(sequence, gotResults);
+      }
+
+      // reset the sequence
+      sequence = [];
+      recording_finished = true;
+    }
+
+  // can only record again when hand is out of frame
+  } else {
+    if (hands.length == 0){
+      recording_finished = false;
+    }
+  }
+}
+
+function drawPoints(){
+  let handpoints = []
+  // iterate through both hands
+  for (let i = 0; i < hands.length; i++) {
+    let hand = hands[i];
+    for (let j = 0; j < hand.keypoints.length; j++) {
+      // access the keypoints in the hand
+      let keypoint = hand.keypoints[j];
+      handpoints.push(keypoint.x,keypoint.y)
+
+      fill(0, 255, 0);
+      noStroke();
+      circle(keypoint.x, keypoint.y, 5);
+    }
+  }
+  // assign to a different variable before clearing
+  const output = handpoints;
+  handpoints = []; 
+  
+  return output;
+}
+
+// Callback function for when handPose outputs data
+function gotHands(results) {
+  // save the output to the hands variable
+  hands = results;
+}
+
+function keyPressed(){
+  if (key == 's'){
+    model.save('hello');
+  }
+  if (key == 'z'){
+    model.saveData();
+  }
+
+  if (key == 't'){
+    model.normalizeData();
+    let options = {
+      epochs: 100
+    }
+    model.train(options,whileTraining,finishedTraining);
+  }
+}
+
+function trainModelAndSave(){
+  model.normalizeData();
+  let options = {
+    epochs: 100
+  }
+  model.train(options,whileTraining,finishedTraining);
+  nameField.value('')
+}
+
+function whileTraining(epoch) {
+  console.log(epoch);
+}
+
+function finishedTraining() {
+  console.log('finished training.');
+  model.save('model');
+}
+
+function gotResults(results){
+  predicted_word = results[0].label
+  console.log(predicted_word)
+  text(predicted_word, 200,200)
+}
+
+function UI(){
+  nameField = createInput('')
+  nameField.attribute('placeholder', 'Type the word to train')
+  nameField.position(110, 500)
+  nameField.size(250)
+
+  instructionP = createP(
+    'I want to train: <br><br> 1.) Type any word you want to pair with a gesture, e.g. "HELLO" <br> 2.) Do the gesture associated to the word, make sure to do it until the points disappear. <br> 3.) Move your hand out of the frame and repeat the gesture, do this multiple times <br> 4.) Do the same for other words e.g. "BYE" <br> 5.) Once all data is collected, press Train and Save<br><br> Tip: have at least 5 datasets for each word'
+  );
+  instructionP.style("width", "640px");
+  dataCountsP = createP(
+    "-> After the gesture a tally will appear here <-"
+  );
+
+  train_but = createButton('Train and Save');
+  train_but.mouseClicked(trainModelAndSave);
+  train_but.style("font-family", "Georgia");
```

Comment:
Hi professor, tried to put the button and initial styles in the HTML file, although the length was not drastically changed, mainly because of the highlighting logic when a button is active, do you think I should remove this feature in favor of code simplicity? thanks!

Reply:
IYou're right that ultimately in total lines of code the benefit might be negligible - but I'd still try to keep those concerns separated, and do styling in HTML/CSS, and merely the event handlers in JS.